smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/PaginationGenerator.java

```diff
@@ -0,0 +1,191 @@
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+
+package software.amazon.smithy.typescript.codegen;
+
+import java.util.Optional;
+
+import software.amazon.smithy.codegen.core.CodegenException;
+import software.amazon.smithy.codegen.core.Symbol;
+import software.amazon.smithy.codegen.core.SymbolProvider;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.knowledge.PaginatedIndex;
+import software.amazon.smithy.model.knowledge.PaginationInfo;
+import software.amazon.smithy.model.shapes.OperationShape;
+import software.amazon.smithy.model.shapes.ServiceShape;
+
+final class PaginationGenerator implements Runnable {
+
+    private final SymbolProvider symbolProvider;
+    private final TypeScriptWriter writer;
+    private final PaginationInfo paginatedInfo;
+
+    private Symbol serviceSymbol;
+    private Symbol operationSymbol;
+    private Symbol inputSymbol;
+    private Symbol outputSymbol;
+
+    private String methodName;
+    private String nonModularServiceName;
+    private String paginationType;
+    private String interfaceLocation;
+
+    PaginationGenerator(Model model,
+                        ServiceShape service,
+                        OperationShape operation,
+                        SymbolProvider symbolProvider,
+                        TypeScriptWriter writer,
+                        String nonModularServiceName) {
+
+        this.symbolProvider = symbolProvider;
+        this.writer = writer;
+
+        this.serviceSymbol = symbolProvider.toSymbol(service);
+        this.operationSymbol = symbolProvider.toSymbol(operation);
+        this.inputSymbol = symbolProvider.toSymbol(operation).expectProperty("inputType", Symbol.class);
+        this.outputSymbol = symbolProvider.toSymbol(operation).expectProperty("outputType", Symbol.class);
+
+        String operationName = operation.getId().getName();
+        this.nonModularServiceName = nonModularServiceName;
+        this.methodName = Character.toLowerCase(operationName.charAt(0))
+                + operationName.substring(1); // e.g. listObjects
+        this.paginationType = this.nonModularServiceName + "PaginationConfiguration";
+        this.interfaceLocation = PaginationGenerator.generateInterfaceFilelocation();
+
+        PaginatedIndex paginatedModel = model.getKnowledge(PaginatedIndex.class);
+        Optional<PaginationInfo> paginationInfo = paginatedModel.getPaginationInfo(service, operation);
+        this.paginatedInfo = paginationInfo.orElseThrow(() -> {
+            return new CodegenException("Expected Paginator to have pagination information.");
+        });
+    }
+
+    @Override
+    public void run() {
+        // Import Service Types
+        writer.addImport(this.operationSymbol.getName(),
```

Comment:
You can remove a lot of the `this` usage in this class. Since there aren't local variables that would make member or method references ambiguous, Java will handle using them automatically.

Reply:
done